src/services/resources-getter.js

```diff
@@ -235,9 +196,11 @@ function ResourcesGetter(model, opts, params) {
 
   this.count = () => getSegmentCondition()
     .then(() => {
-      const query = getRecordsQuery();
-
-      return hasRelationshipFilter() ? exec(query).then(records => records.length) : count(query);
+      const jsonQuery = getRecordsQuery();
+      jsonQuery.push({ $group: { _id: null, count: { $sum: 1 } } });
+      jsonQuery.push({ $project: { _id: 0 } });
+      return model.aggregate(jsonQuery)
+        .then(result => result[0].count);
```

Comment:
You have a test error on the count function.
Take the time to fix it 
<img width="1062" alt="Screenshot 2019-07-29 at 08 35 22" src="https://user-images.githubusercontent.com/664261/62026642-d6ba5280-b1db-11e9-9494-a8420d7e57c4.png">

Reply:
I think you should take the time to write a few tests (unless Valentin already wrote them).
Because this part is not really tested and this is an important code change.

Reply:
Valentin added a lot of test but on the new filters. But I may create for the old one if you want

Reply:
The previous comment you made fix the test